scripts/post_autosave.js

```diff
@@ -0,0 +1,164 @@
+﻿/*
+ TODO:
+
+ Trigger removal of cached posts off successful posting instead of form submission (in case something goes wrong)
+*/
+settingsLoadedEvent.addHandler(function() {
+    if (getSetting("enabled_scripts").contains("post_autosave")) {
+        PostAutosave = {
+            CACHE_SETTING_NAME: 'autosaved_posts',
+
+            // Max number of posts cached
+            CACHE_SIZE: 10,
+
+            // The amount of time (ms) after typing before posts are cached 
+            CACHE_TIME: 500,
+
+            // Local copy of the cache to use within a single page view
+            cache: [],
```

Comment:
Fair enough, then instead of iterating everywhere, you can just do something like:

```
var post = posts.find(post => post.id = something)
if (post) {
  //update
else
  //add